src/com/palmergames/bukkit/towny/db/TownyDatabaseHandler.java

```diff
@@ -1562,4 +1566,77 @@ public void mergeNation(Nation succumbingNation, Nation prevailingNation) {
 		}
 		lock.unlock();
 	}
+
+	/**
+	 * Merges the mergeFrom town into the mergeInto town.
+	 * @param mergeInto The town that the other town merges into.
+	 * @param mergeFrom The town that will be deleted.
+	 */
+	public void mergeTown(Town mergeInto, Town mergeFrom) {
+		if (TownyEconomyHandler.isActive()) {
+			try {
+				if (mergeFrom.getAccount().getHoldingBalance() > 0)
+					mergeFrom.getAccount().payTo(mergeFrom.getAccount().getHoldingBalance(), mergeInto, Translation.of("msg_town_merge_transaction_reason"));
+			} catch (EconomyException ignored) {}
+		}
+
+		lock.lock();
+		String mayorName = mergeFrom.getMayor().getName();
+		List<Location> jails = new ArrayList<Location>(mergeFrom.getAllJailSpawns());
+		List<Location> outposts = new ArrayList<Location>(mergeFrom.getAllOutpostSpawns());
+
```

Comment:
The succumbin town will have to pass on the following values to the remaining town:

- bonusBlocks
- purchasedBlocks